dpnp/dpnp_iface_linearalgebra.py

```diff
@@ -98,6 +98,9 @@ def dot(x1, x2, **kwargs):
         if(x1_desc.ndim == 0 and x2_desc.strides is not None
                 or x2_desc.ndim == 0 and x1_desc.strides is not None):
             pass
+        elif (x1_desc.ndim >= 1 and x2_desc.ndim > 1 and x1_desc.shape[-1] != x2_desc.shape[-2]
+                or x2_desc.ndim == 1 and x1_desc.shape[-1] != x2_desc.shape[0]):
```

Comment:
Let's separate the condition to 2 ones adding one more `elif`. Then for the second condition add check `x1_desc.ndim > 0`.